content/doc/book/pipeline/shared-libraries.adoc

```diff
@@ -194,6 +194,54 @@ A link to the example repository used is also provided in the link:https://youtu
 
 video::eV7roTXrEqg[youtube,width=800,height=420]
 
+=== Statically loading evaluated versions of libraries
+
+As of currently unreleased version of the _Pipeline: Shared Groovy Libraries_ plugin,
+with features proposed in
+link:https://github.com/jenkinsci/pipeline-groovy-lib-plugin/pull/19[PR #19]
+(link:https://issues.jenkins.io/browse/JENKINS-69731[issue JENKINS-69731])
+there are new options for the `@Library` annotation with limited run-time evaluation
+of the version specified for the library. In order to use these features, you should
+explicitly allow them in library configuration (e.g. Global Jenkins configuration of
+your instance).
+
+While the markup for these features is intentionally similar to what you would write
+for Groovy code to simplify maintenance, keep in mind that the version tokens are
+handled by the plugin before it would load libraries and compile the pipeline script.
+
+One addressed use-case regards loading of "same version of library as the version of
+Pipeline script from SCM" by specifying literally +`${BRANCH_NAME}`+ as the library
+version in the pipeline script, e.g.:
+
+[source,groovy]
+----
+@Library('my-shared-library@${BRANCH_NAME}') _
+----
+
+The plugin would try several methods to determine the branch name to use, check if
+such "version" is accepted by SCM source of the library, and fall back to default
+version if not.
+
+This was verified to work for:
+* MultiBranch Project pipelines both with "Modern SCM" and with "Single repository & branch"
+  (in case of the latter, the "name" you specify for the source is what matters for MBP more
+  than the actual SCM branch);
+* Pipelines backed by a single SCM source with a "Branch Specifier" like `*/main`;
+* "In-line" Pipelines not backed by an SCM (just fall back to default version).
+
+Another addressed use-case regards loading of some version of library specified by
```

Comment:
Possibly simpler phrasing?
```suggestion
The plugin can also load a library version specified by
```

Reply:
Not sure about this change (alone) since an earlier paragraph stressed that "one use-case is..." so here "another"...